stdlib/influxdata/influxdb/to.go

```diff
@@ -1,15 +1,604 @@
+}
+
+// Copy clones the procedure spec for `to` flux function.
+func (o *ToProcedureSpec) Copy() plan.ProcedureSpec {
+	s := o.Spec
+	res := &ToProcedureSpec{
+		Spec: &ToOpSpec{
+			Bucket:            s.Bucket,
+			BucketID:          s.BucketID,
+			Org:               s.Org,
+			OrgID:             s.OrgID,
+			Host:              s.Host,
+			Token:             s.Token,
+			TimeColumn:        s.TimeColumn,
+			MeasurementColumn: s.MeasurementColumn,
+			TagColumns:        append([]string(nil), s.TagColumns...),
+			FieldFn:           s.FieldFn.Copy(),
+		},
+	}
+	return res
+}
+
+func newToProcedure(qs flux.OperationSpec, a plan.Administration) (plan.ProcedureSpec, error) {
+	spec, ok := qs.(*ToOpSpec)
+	if !ok && spec != nil {
+		return nil, &flux.Error{
+			Code: codes.Internal,
+			Msg:  fmt.Sprintf("invalid spec type %T", qs),
+		}
+	}
+	return &ToProcedureSpec{Spec: spec}, nil
+}
+
+// argsReader is an interface for OperationSpec that have the same method to read args.
+type argsReader interface {
+	flux.OperationSpec
+	ReadArgs(args flux.Arguments) error
+}
+
+// ReadArgs reads the args from flux.Arguments into the op spec
+func (o *ToOpSpec) ReadArgs(args flux.Arguments) error {
+	var err error
+	var ok bool
+
+	if o.Bucket, ok, _ = args.GetString("bucket"); !ok {
+		if o.BucketID, err = args.GetRequiredString("bucketID"); err != nil {
+			return err
+		}
+	} else if o.BucketID, ok, _ = args.GetString("bucketID"); ok {
+		return &flux.Error{
+			Code: codes.Invalid,
+			Msg:  "cannot provide both `bucket` and `bucketID` parameters to the `to` function",
+		}
+	}
+
+	if o.Org, ok, _ = args.GetString("org"); !ok {
+		if o.OrgID, _, err = args.GetString("orgID"); err != nil {
+			return err
+		}
+	} else if o.OrgID, ok, _ = args.GetString("orgID"); ok {
+		return &flux.Error{
+			Code: codes.Invalid,
+			Msg:  "cannot provide both `org` and `orgID` parameters to the `to` function",
+		}
+	}
+
+	if o.Host, ok, _ = args.GetString("host"); ok {
+		if o.Token, err = args.GetRequiredString("token"); err != nil {
+			return err
+		}
+	}
+
+	if o.TimeColumn, ok, _ = args.GetString("timeColumn"); !ok {
+		o.TimeColumn = execute.DefaultTimeColLabel
+	}
+
+	if o.MeasurementColumn, ok, _ = args.GetString("measurementColumn"); !ok {
+		o.MeasurementColumn = defaultMeasurementColLabel
+	}
+
+	if tags, ok, _ := args.GetArray("tagColumns", semantic.String); ok {
+		o.TagColumns = make([]string, tags.Len())
+		tags.Sort(func(i, j values.Value) bool {
+			return i.Str() < j.Str()
+		})
+		tags.Range(func(i int, v values.Value) {
+			o.TagColumns[i] = v.Str()
+		})
+	}
+
+	if fieldFn, ok, _ := args.GetFunction("fieldFn"); ok {
+		if o.FieldFn, err = interpreter.ResolveFunction(fieldFn); err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+func createToOpSpec(args flux.Arguments, a *flux.Administration) (flux.OperationSpec, error) {
+	if err := a.AddParentFromArgs(args); err != nil {
+		return nil, err
+	}
+
+	_, httpOK, err := args.GetString("url")
```

Comment:
I don't think that url or brokers are valid arguments for this to function. I assume this was probably when `to()` was going to be an all-encompassing function that included choosing the correct implementation automatically. Originally, from was going to take multiple arguments and choose which underlying from to call depending on those arguments.

With that, I think you can remove the entirety of the kafka related stuff here and either do this:

```go
s := &ToOpSpec{}
if err := s.ReadArgs(args); err != nil {
    return nil, err
}
return s, nil
```

Or you can remove that `ReadArgs` function altogether and replace the implementation here with it.

Reply:
I kept ReadArgs - I think it reads a bit more nicely.